pkg/block/s3/adapter.go

```diff
@@ -120,6 +127,7 @@ func NewAdapter(ctx context.Context, params params.S3, opts ...AdapterOption) (*
 	a := &Adapter{
```

Comment:
What about `adapter.GetPresignUploadPartURL` endpoint override? I only see it in `adapter.GetPreSignedURL`